CondCore/ESSources/plugins/CondDBESSource.cc

```diff
@@ -741,6 +738,45 @@ void CondDBESSource::fillTagCollectionFromDB(const std::vector<std::string>& con
   }
 }
 
+void CondDBESSource::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  edm::ParameterSetDescription dbParams;
+  dbParams.addUntracked<std::string>("authenticationPath", "");
+  dbParams.addUntracked<int>("authenticationSystem", 0);
+  dbParams.addUntracked<std::string>("security", "");
+  dbParams.addUntracked<int>("messageLevel", 0);
+  dbParams.addUntracked<int>("connectionTimeout", 0);
+  desc.add("DBParameters", dbParams);
+
+  desc.add<std::string>("connect", std::string("frontier://FrontierProd/CMS_CONDITIONS"));
+  desc.add<std::string>("globaltag", "");
+  desc.add<std::string>("snapshotTime", "");
+  desc.addUntracked<std::string>("frontierKey", "");
+
+  edm::ParameterSetDescription toGetDesc;
+  toGetDesc.add<std::string>("record", "");
```

Comment:
As an empty record name will just cause an exception, I'd suggest not adding a default. The framework will _do the right thing_ and still provide the python with the fact that `record` is a valid key and the type is `cms.string`.

Reply:
sorry that doesn't work for confDB purpose. we need a default to be able to parse it.

Reply:
Sorry, why is "" not allowed as value for a string parameter called record?

Reply:
If the VPSet toGet requires an entry PSet, its record parameter will be set to a proper value (within ConfDb).

Reply:
>  It would be nice if at some point the CondDB could be updated to work with the improved features of the configuration system.

probably in Run4